### ISSUE TITLE
Initialized bug in MAGNets causing issues with model loading 

### ISSUE THREAD

**Issue by Aphoh** [david-knigge/ccnn#14]
Hello @david-knigge! I've found a bug in `ckconv.py` that will cause a trained model which is loaded via state dict and checkpoint to have some of it's weights re-initialized at inference time. 

Specifically this line:
```python
        self.register_buffer("initialized", torch.zeros(1).bool(), persistent=False)
``` 
And in the `chang_initialization` method:
```python
        if not self.initialized[0] and self.chang_initialize:
```

Using `register_buffer` with `persistent=False` for this is incorrect, as this means `self.initialized` is not saved in the state dict. This will cause a freshly created model object with it's state dict loaded to have `self.initialized[0] = [False]`, causing weights to be re-initialized when inference is done. Instead, this should be changed to `persistent=True` and then it should get saved in the state dict.

This bug was **really** hard to figure out (you don't expect your weights to change when running inference 😆). I think this might also be the reason why in `main.py` you have somewhat weird code for loading in models. Hopefully this helps anyone else building on this repository!

**Comment by onderakacik** [comment]
@Aphoh Thank you very much for pointing this out! I was not able to get the same test accuracy when I loaded the model from the checkpoint. Turns out that bug is the reason. 😄

**Comment by dwromero** [comment]
Sorry for the long wait. We'll start modernizing this repo in the following weeks. We'll keep you up-to-date :)
